File: ios/PersonalOSApp/Design/POSTheme.swift
import SwiftUI

enum POSTheme {
    static let background = Color(red: 249 / 255, green: 247 / 255, blue: 242 / 255)
    static let card = Color(red: 255 / 255, green: 253 / 255, blue: 250 / 255)
    static let ink = Color(red: 38 / 255, green: 32 / 255, blue: 28 / 255)
    static let foreground = ink
    static let primary = Color(red: 176 / 255, green: 52 / 255, blue: 68 / 255)
    static let primaryDark = Color(red: 140 / 255, green: 38 / 255, blue: 52 / 255)
    static let muted = Color(red: 118 / 255, green: 108 / 255, blue: 98 / 255)
    static let border = Color(red: 228 / 255, green: 220 / 255, blue: 210 / 255)
    static let paperLine = Color(red: 220 / 255, green: 210 / 255, blue: 198 / 255)
    static let paperHighlight = Color.white
    static let paperShadow = Color(red: 90 / 255, green: 70 / 255, blue: 55 / 255)
    static let success = Color(red: 46 / 255, green: 120 / 255, blue: 78 / 255)
    static let successBg = Color(red: 232 / 255, green: 246 / 255, blue: 236 / 255)
    static let focus = Color(red: 58 / 255, green: 102 / 255, blue: 88 / 255)

    static let cardRadius: CGFloat = 20
    static let tabBarHeight: CGFloat = 60
}

extension Font {
    static func posDisplay(_ size: CGFloat, weight: Font.Weight = .semibold) -> Font {
        .system(size: size, weight: weight, design: .serif)
    }

    static func posLabel(_ size: CGFloat = 11) -> Font {
        .system(size: size, weight: .medium)
    }

    static func posCaps(_ size: CGFloat = 10) -> Font {
        .system(size: size, weight: .semibold)
    }
}

File: android/app/src/main/java/com/personalos/mobile/ui/theme/PosTheme.kt
package com.personalos.mobile.ui.theme

import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.lightColorScheme
import androidx.compose.runtime.Composable
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.TextStyle
import androidx.compose.ui.text.font.FontFamily
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp

object PosTheme {
    val Background = Color(0xFFF9F7F2)
    val Card = Color(0xFFFFFDFA)
    val Ink = Color(0xFF26201C)
    val Primary = Color(0xFFB03444)
    val PrimaryDark = Color(0xFF8C2634)
    val Muted = Color(0xFF766C62)
    val Border = Color(0xFFE4DCD2)
    val PaperLine = Color(0xFFDCD2C6)
    val PaperHighlight = Color.White
    val PaperShadow = Color(0xFF5A4637)
    val Focus = Color(0xFF3A6658)
    val Success = Color(0xFF2E784E)
    val SuccessBg = Color(0xFFE8F6EC)
    val Error = Color(0xFFB03444)
    val CardRadius = 20.dp
    val TabBarHeight = 60.dp
}

private val LightColors = lightColorScheme(
    primary = PosTheme.PrimaryDark,
    onPrimary = Color.White,
    background = PosTheme.Background,
    onBackground = PosTheme.Ink,
    surface = PosTheme.Card,
    onSurface = PosTheme.Ink,
    outline = PosTheme.Border,
)

@Composable
fun PersonalOSTheme(content: @Composable () -> Unit) {
    MaterialTheme(
        colorScheme = LightColors,
        typography = androidx.compose.material3.Typography(
            headlineLarge = TextStyle(
                fontFamily = FontFamily.Serif,
                fontWeight = FontWeight.SemiBold,
                fontSize = 28.sp,
            ),
            titleMedium = TextStyle(
                fontFamily = FontFamily.Serif,
                fontWeight = FontWeight.SemiBold,
                fontSize = 17.sp,
            ),
            bodyMedium = TextStyle(fontSize = 14.sp),
            labelSmall = TextStyle(fontSize = 10.sp, fontWeight = FontWeight.SemiBold),
        ),
        content = content,
    )
}

fun posDisplay(size: Float = 28f, weight: FontWeight = FontWeight.SemiBold) = TextStyle(
    fontFamily = FontFamily.Serif,
    fontWeight = weight,
    fontSize = size.sp,
)

fun posLabel(size: Float = 11f) = TextStyle(
    fontWeight = FontWeight.Medium,
    fontSize = size.sp,
)

fun posCaps(size: Float = 10f) = TextStyle(
    fontWeight = FontWeight.SemiBold,
    fontSize = size.sp,
)
